### PR TITLE
Update build script to export Vue SVG files

### PR DESCRIPTION
**Pull request by hynes-dialpad**
Per our discussion in today's meeting, this updates `gulpfile.js` so that it also exports the SVG files into their own Vue component files.

Each SVG is wrapped in a `<template>` tag.

Also the SVG file names are renamed. Currently the `SVG` files are lowercase, separated with single dashes. The script name changes this to PascalCasing, prepending the word "Icon" to the beginning of every `.vue` file. An example:

```
IconAdd.vue
IconAttachFile.vue
```

### CODE REVIEW

**Review by dbecherdialpad — APPROVED** [review]
Awesome! Are these all in the documentation now?

**Comment by hynes-dialpad** [comment]
@dbecherdialpad Ah, no. I should add the bit about Vue now.

File: docs/_data/icons.yml [hunk 1]
```diff
@@ -1,264 +1,396 @@
...
+  vue: IconArrowDropDown
 - icon: Arrow, Drop Up
   file: arrow-drop-up
+  vue: IconArrowDropUp
 - icon: Arrow, Forward (iOS Style)
   file: arrow-forward--ios
+  vue: IconArrowForwardIos
 - icon: Arrow, Forward
   file: arrow-forward
+  vue: IconArrow Forward
```
Comment (dbecherdialpad) [inline docs/_data/icons.yml]: space

File: lib/build/less/dialtone-globals.less [hunk 2]
```diff
@@ -14,3 +14,41 @@ body {
...
+
+    width: 3.6rem;
+    height: 3.6rem;
+}
+.d-svg--xl {
+    .d-svg();
+
+    width: 4.8rem;
+    height: 4.8rem;
+}
```
Comment (dbecherdialpad) [inline lib/build/less/dialtone-globals.less:98]: while you're at it maybe should think about adding `variables/transitions.less` to this globals file? It similarly needs to be globally defined.

File: lib/dist/css/dialtone.css [hunk 3]
```diff
@@ -6251,3 +6248,36 @@ body {
...
   font-size: 1.6rem;
 }
+.d-svg {
+  width: 2.4rem;
+  height: 2.4rem;
+  fill: currentColor;
+}
+.d-svg--xs {
+  width: 2.4rem;
+  width: 1.4rem;
```
Comment (dbecherdialpad) [inline lib/dist/css/dialtone.css:6026]: weird what's going on with width/height being defined twice?

**Review by dbecherdialpad — APPROVED** [review]
this is awesome, thanks for your hard work on getting it done so quickly!

File: docs/_data/icons.yml (hunk 1 above)
Comment (hynes-dialpad) [reply docs/_data/icons.yml]: This has been fixed.

File: lib/dist/css/dialtone.css (hunk 3 above)
Comment (hynes-dialpad) [reply lib/dist/css/dialtone.css:6026]: I think it is from the fact its inheriting everything from d-svg.

**Pull request merged**